### DISCUSSION TITLE
how do i actually use this?

### DISCUSSION THREAD

**Discussion by giveyouup** [jazzyisj/unavailable-entities-sensor#28] [Q&A] [answered]
followed the  install instructions and now i'm stuck. How do i actually use this? Does it show up under devices or entities? THis is probably a  noob question so my apologies. I am pretty sure i have it "installed" but i don't see it anywhere in HA after a quick reload. I want to see what the sensor found, how do i view its output?

**Comment by jazzyisj** [comment]
Where do you want to see this sensor?  Do you want a notification?  Do you want to see it in the front end somewhere?  There is an example automation to send a notification in the package to get you started if that is what you want.

[Here is an example from my config using the sensor in the front end](https://github.com/jazzyisj/home-assistant-config/blob/377cc5d10a644f291d85ea60ee132b89b03576ed/ui/card/hass/device_monitor.yaml#L93).

You can check to see if the sensor is working in the developer tools.
![image](https://user-images.githubusercontent.com/44064559/222483739-d68f170b-30a2-4088-87c8-05ef79d5427e.png)

**Reply by giveyouup** [reply]
ahh i see. What i was picturing was seeing this on my dashboard. a card with # of offline devices, then perhaps a pop-up card  to see details with names etc. Would I have to make my own custom card for this? I tried using your code from above example but it's giving me errors

**Reply by jazzyisj** [reply, marked as answer]
Edit: Popups are done with [browser-mod](https://github.com/thomasloven/hass-browser_mod).  All of these components are written by [Thomas Loven](https://github.com/thomasloven/) btw, who is a longstanding prolific contributor to HASS, especially the front end.

My code does use some custom components.  The sensor entity is a "[fold-entity-row](https://github.com/thomasloven/lovelace-fold-entity-row)" and is populated with "[auto-entities](https://github.com/thomasloven/lovelace-auto-entities)" which is another custom component.

It looks like this collapsed in an entities card.

<img width="307" alt="image" src="https://user-images.githubusercontent.com/44064559/222590788-668fafd1-3e78-429c-ba20-8f432840316b.png">

And like this expanded.

<img width="295" alt="image" src="https://user-images.githubusercontent.com/44064559/222591086-c8a87f07-eaf8-4846-9932-b08d9d4a4271.png">

To keep it real simple, you can add the sensor as a basic entity on an entities card as you would any other entity.  The unavailable entities are listed in the more info popup.

<img width="418" alt="image" src="https://user-images.githubusercontent.com/44064559/222591786-8715b8c3-20c1-4bd0-bdd7-e75cbca88517.png">

You could use the sample automation in the package if you want to keep it simple.  It will give you a persistent notification like this.

<img width="379" alt="image" src="https://user-images.githubusercontent.com/44064559/222592965-8c7012ff-68e8-4d47-bf24-d62c7453b55d.png">

**Reply by giveyouup** [reply]
Thanks man, I got it working thanks to your help. Very nice to have this

**Comment by giveyouup** [comment]
hello again!
Now having a bit of trouble filtering out devices I don't want. Code below gives me a "line break expected" error and "string does not match pattern" error. Probably missing something like a bracket or line but i've spent hours trying to figure it out but just get one error after another. Even tried to feed the code to chatGPT to have it debug it but it thinks it knows what it's doing but it doesn't 
```
template:
  - sensor:
      - name: "Unavailable Entities"
        unique_id: unavailable_entities
        icon: "{{ iif(states(this.entity_id)|int(-1) > 0,'mdi:alert-circle','mdi:check-circle') }}"
        state: >
          {% set entities = state_attr(this.entity_id,'entity_id') %}
          {{ entities|count if entities != none else none }}
        attributes:
          entity_id: >
            {% set ignore_seconds = 60 %}
            {% set ignored = state_attr('group.ignored_unavailable_entities','entity_id') %}
            {% set ignore_ts = (now().timestamp() - ignore_seconds)|as_datetime %}
            {% set entities = states|rejectattr('domain','eq','group')
                |rejectattr('last_changed','ge',ignore_ts)
                |selectattr('state','in',['unavailable','unknown']) %}
            {% set entities =  entities|rejectattr('entity_id','in',ignored)
                if ignored != none else entities %}
            {{ entities|map(attribute='entity_id')|list }}

group:
  ignored_unavailable_entities:
    entities:
      - |rejectattr('entity_id','search','echo')
```



only the last block of code is changed,

**Reply by jazzyisj** [reply]
You can only add jinja where a template is expected.   Defining a group is done with YAML and must be a list of individual entities.

https://www.home-assistant.io/integrations/group/#yaml-configuration

The code you are trying to use needs to be part of the template portion.

```
          entity_id: >
            {% set ignore_seconds = 60 %}
            {% set ignored = state_attr('group.ignored_unavailable_entities','entity_id') %}
            {% set ignore_ts = (now().timestamp() - ignore_seconds)|as_datetime %}
            {% set entities = states|rejectattr('domain','eq','group')
                |rejectattr('last_changed','ge',ignore_ts)
                |rejectattr('entity_id','search','echo')
                |selectattr('state','in',['unavailable','unknown']) %}
            {% set entities =  entities|rejectattr('entity_id','in',ignored)
                if ignored != none else entities %}
            {{ entities|map(attribute='entity_id')|list }}
```

**Reply by giveyouup** [reply]
a real facepalm moment. Thank you, i did not know that
